core/common/test/DateTimePeriodTest.kt

```diff
@@ -29,8 +29,43 @@ class DateTimePeriodTest {
         assertEquals("-P1DT1H", DateTimePeriod(days = -1, hours = -1).toString())
         assertEquals("-P1M", DateTimePeriod(months = -1).toString())
 
-        assertEquals("P-1Y-2M-3DT-4H-5M0.500000000S",
+        assertEquals("-P1Y2M3DT4H4M59.500000000S",
                 DateTimePeriod(years = -1, months = -2, days = -3, hours = -4, minutes = -5, seconds = 0, nanoseconds = 500_000_000).toString())
+
+        assertEquals("PT277H46M39.999999999S", DateTimePeriod(nanoseconds = 999_999_999_999_999L).toString())
+        assertEquals("PT0.999999999S", DateTimePeriod(seconds = 1, nanoseconds = -1L).toString())
+        assertEquals("-PT0.000000001S", DateTimePeriod(nanoseconds = -1L).toString())
+        assertEquals("P1DT-0.000000001S", DateTimePeriod(days = 1, nanoseconds = -1L).toString())
+        assertEquals("-PT0.999999999S", DateTimePeriod(seconds = -1, nanoseconds = 1L).toString())
+        assertEquals("P1DT-0.999999999S", DateTimePeriod(days = 1, seconds = -1, nanoseconds = 1L).toString())
+    }
+
+    @Test
+    fun parseIsoString() {
+        assertEquals(DateTimePeriod(years = 1), DateTimePeriod.parse("P1Y"))
```

Comment:
Better to assert the expected property values rather than equality as a whole in this test. Because otherwise it's unclear what values these properties would get as a result of parsing.

Reply:
You raise a good point, though I think that the values of properties and the results of parsing are separate concerns, so I added another test to check normalization.